File: my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_no_oop")
    client = node.create_client(AddTwoInts, "add_two_ints")

    while not client.wait_for_service(1.0):
        node.get_logger().warn("Waiting for Server...")

    request = AddTwoInts.Request()
    request.a = 10
    request.b = 12

    future_obj = client.call_async(request)
    rclpy.spin_until_future_complete(node, future_obj)
    
    try:
        response = future_obj.result()
        node.get_logger().info(str(request.a) + " + " + str(request.b) + " = " + str(response.sum)) 

    except Exception as e:
        node.get_logger().error("Service call failed %r" % (e,))
  
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop");
    // creating client node
    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    // wait for the server to be ready
    while (!client->wait_for_service(std::chrono::seconds(1)))
    {
        RCLCPP_WARN(node->get_logger(),"Waiting for the server to be up...");
    }
    // creating the request
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 3;
    request->b = 5;
    // send the request asynchronously from the client to the server
    auto future_obj = client->async_send_request(request);
    if (rclcpp::spin_until_future_complete(node, future_obj) == rclcpp::FutureReturnCode::SUCCESS)
    {
        RCLCPP_INFO(node->get_logger(), "%d + %d = %d", (int)request->a, (int)request->b, 
                                                        (int)future_obj.get()->sum);
    }

    else{
        RCLCPP_ERROR(node->get_logger(), "Error while calling service");
    }

    rclcpp::shutdown();
    return 0;
}